plugins/org.eclipse.elk.alg.common/src/org/eclipse/elk/alg/common/nodespacing/NodeMarginCalculator.java

```diff
@@ -185,29 +176,33 @@ private void processNode(final NodeAdapter<?> node, final double labelSpacing) {
             }
             
             // End labels of edges connected to the port
+            // TODO cds: layered calls #excludeEdgeHeadTailLabels, is this ever active?
```

Comment:
ELK Layered does its own end label processing in `EndLabelPreprocessor` and `EndLabelPostprocessor`. I think this code wasmeant for algorithms that do not implement their own sophisticated end label processing.